### DISCUSSION TITLE
Add more colors for TopBottomShader

### DISCUSSION THREAD

**Discussion by calmwalija** [patrykandpatrick/vico#681] [Ideas]
I suggest if you can have the option to add more than two colors for [TopBottomShader](https://patrykandpatrick.com/vico/api/2.0.0-alpha.19/core/com.patrykandpatrick.vico.core.common.shader/-top-bottom-shader/) it would be great.

**Comment by calmwalija** [comment]
cc: @yago1994

**Comment by patrickmichalik** [comment]
Hello! Thanks for the suggestion. `LineCartesianLayer` has special support for `TopBottomShader`, but you can use `TopBottomShader` wherever a `DynamicShader` is accepted. Just to confirm, is this in the context of `LineCartesianLayer`, or are you referring to `TopBottomShader` more generally?

**Comment by calmwalija** [comment]
![Asset 2@4x-100](https://github.com/patrykandpatrick/vico/assets/42068747/f829906f-81da-4ab7-9f48-a0524c755420)

Something like that, you should be able to add more colors not only just two to the `TopBottomShader`

**Reply by patrickmichalik** [reply]
Thanks for clarifying. This is a planned enhancement. A similar use case was described in discussion #483, and the API that we plan on introducing will cover both that and the use case you’ve described. I’ll thus be closing this discussion as a duplicate of discussion #483, where updates will be shared. Cheers!

**Reply by calmwalija** [reply]
Alright, thank you!
